### ISSUE TITLE
Unfolding of subject header removes consecutive spaces

### ISSUE THREAD

**Issue by aselder** [mikel/mail#980]
``` ruby
s = "Your Order  with ABC"
Mail::Field.new('subject', s, 'UTF-8')
f.to_s == s # => false
f.to_s # => "Your Order with ABC"
```

I've narrowed it down to line 236 in field.rb:

``` ruby
s.gsub(/[\r\n \t]+/m, ' ') # => "Your Order with ABC"
```

Is there any reason to strip turn multiple spaces in a subject field. If nothing else, this violates the principle of least surprise.

I was writing a spec to test a mailer and the subject I sent in wasn't what I got back out.

**Comment by jeremy** [comment]
Agree it's odd. What does git history indicate, @aselder?

**Comment by aselder** [comment]
@jeremy This was last changed 2/1/13 in 72befdc4dab3e6e288ce226a7da2aa474cf5be83, with the comment "This has little effect on the specs, but on my header reading example it
makes about a 10x performance difference"

So it looks like this was a performance enhancement that led to a regression. The only reason I'm discovering it is we're upgrade from Rails 3 (mail 2.5.x) to Rails 4 (mail 2.6.x). All of a sudden a bunch of specs started failing.

**Comment by aselder** [comment]
The comment says "Unfolding is accomplished by simply removing any CRLF that is immediately followed by WSP."

How ever this is not what the regex does. Based on the comment, I can probably correct the regex.

**Comment by vihai** [comment]
This issue has major consequences when it comes to S/MIME signing the messages as the headers contents are not preserved and the signature is made invalid.

**Comment by jeremy** [comment]
Fixed by #982
